odoo/osv/expression.py

```diff
@@ -968,8 +968,12 @@ def push_result(query, params):
                         right = tuple(right)
 
                     unaccent = self._unaccent(field) if sql_operator.endswith('like') else lambda x: x
-
-                    left = unaccent(model._generate_translated_field(alias, left, self.query))
+                    if need_wildcard:
+                        # Inactive languages are not removed in DB and can be searched by using this function
+                        left = unaccent(f'jsonb_path_query_array("{alias}"."{left}", \'$.*\')::text')
```

Comment:
Same note as above, `array_to_text` might be a better conversion function than `::text`.